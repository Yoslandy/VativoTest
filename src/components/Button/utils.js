export const sizes = {
    small: "h-8 w-[100px]",
    big: "h-12 w-full",
}

export const types = {
    primary:
        "bg-primary-60 text-white active:bg-primary-100 disabled:bg-neutral-gray-70",
    secondary:
        "bg-transparent bg-primary-60 border-2 border-primary-60 text-primary-60 active:border-primary-100 active:text-primary-100 hover:border-primary-100 hover:text-primary-100 disabled:transparent disabled:text-neutral-gray-70 disabled:border-neutral-gray-70 disabled:shadow-none",
    warning: "bg-red-500 text-white active:bg-primary-100",
    success: "bg-emerald-500 text-white active:bg-primary-100",
    outline:
        "bg-white text-primary-60 border-2 border-primary-60 active:bg-primary-100",
    enabled: "bg-primary-100 text-white",
    tertiary: "bg-primary-100 text-white",
}
