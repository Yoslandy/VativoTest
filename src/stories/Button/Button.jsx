import React from "react"
import PropTypes from "prop-types"

import Button from "../../components/Button"

export default function CustomButton({ label, onClick, type }) {
    return <Button text={label} onClick={onClick} type={type} />
}

CustomButton.propTypes = {
    label: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    type: PropTypes.oneOf(["primary", "secondary"]),
}

CustomButton.defaultProps = {
    type: "primary",
}
