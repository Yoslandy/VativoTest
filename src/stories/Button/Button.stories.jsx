import React from "react"

import "@fortawesome/fontawesome-free/css/all.min.css"
import "../../assets/styles/build.css"

import Button from "./Button"

export default {
    title: "Example/Button",
    component: Button,
    argTypes: {
        label: "Holi",
        type: "primary",
        onClick: { action: "clicked" },
    },
}

function Template(args) {
    return <Button {...args} />
}

export const Primary = Template.bind({})
Primary.args = {
    type: "primary",
    label: "Primary",
}

export const Secondary = Template.bind({})
Secondary.args = {
    type: "secondary",
    label: "Secondary",
}

export const Warning = Template.bind({})
Warning.args = {
    type: "warning",
    label: "Warning",
}

export const Success = Template.bind({})
Success.args = {
    type: "success",
    label: "Success",
}

export const Outline = Template.bind({})
Outline.args = {
    type: "outline",
    label: "Outline",
}

export const Default = Template.bind({})
Default.args = {
    label: "Default",
}
