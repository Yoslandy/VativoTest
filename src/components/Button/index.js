import React from "react"
import PropTypes from "prop-types"
import { sizes, types } from "./utils"

export default function Button(props) {
    const {
        text,
        onClick,
        type,
        className,
        size,
        icon,
        iconRigth,
        disabled,
        submit,
    } = props

    return (
        <button
            className={`button ${sizes[size]} ${types[type]} ${className}`}
            type={submit ? "submit" : "button"}
            onClick={onClick}
            disabled={disabled}
        >
            {icon} {text} {iconRigth}
        </button>
    )
}

Button.propTypes = {
    text: PropTypes.string,
    onClick: PropTypes.func,
    size: PropTypes.oneOf(["small", "big"]),
    className: PropTypes.string,
    type: PropTypes.oneOf([
        "primary",
        "secondary",
        "warning",
        "outline",
        "success",
        "",
    ]),
    icon: PropTypes.node,
    iconRigth: PropTypes.node,
    disabled: PropTypes.bool,
    submit: PropTypes.bool,
}

Button.defaultProps = {
    text: "My Button",
    onClick: () => {},
    type: "primary",
    className: "",
    size: "small",
    icon: null,
    iconRigth: null,
    disabled: false,
    submit: false,
}
