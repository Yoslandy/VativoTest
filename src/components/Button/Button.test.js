import React from "react"
import { MemoryRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import Button from "./index"

test("Renders Button form correctly", () => {
    const component = renderer.create(
        <MemoryRouter>
            <Button type="primary" />
        </MemoryRouter>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
