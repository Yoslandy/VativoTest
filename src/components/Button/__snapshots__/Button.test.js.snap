// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Button form correctly 1`] = `
<button
  className="button bg-primary-60 text-white active:bg-primary-100 h-8 w-[100px] bg-primary-100 "
  disabled={false}
  onClick={[Function]}
  type="button"
>
   
  My Button
   
</button>
`;
